party.go

```diff
@@ -0,0 +1,84 @@
+package pokemonbattlelib
+
+import (
+	"errors"
+	"log"
+)
+
+// A Pokemon party. Can hold up to 6 Pokemon. Also manages how many pokemon are out on the battlefield.
+type party struct {
+	Agent         *Agent     // The agent that has control over this party
+	pokemon       []*Pokemon // The Pokemon in the party
+	activePokemon []*Pokemon // The pokemon in the party that are out on the battlefield
+	team          int        // The team that this party belongs to
+}
```

Comment:
Suppose a user wanted to do multiple battles in a row with the same party of pokemon. Would the user have to use their own type to keep track of the party, and then convert it into this one when they need to put it into a new battle?

In that scenario, we can't guarantee that `team` will remain valid if this party is directly reused. Nor can we guarantee that `Agent` doesn't have state that would make it behave differently than if it was freshly created.

Reply:
Since the party holds direct references to the Pokemon I was concerned about immutability during battle state so I wanted to keep it unexported. There's probably better solutions here, but we potentially can have a party reset helper similar to a PokeCenter (items are tricky in that case). Depending on #61 it might become trivial to loop through battles from some initial setup.

I still think teams should be managed by the user, but each battle/round asserts that there is a valid team configuration.